src/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Packages/PackageUpdaterStep.cs

```diff
@@ -75,7 +78,7 @@ protected override async Task<UpgradeStepInitializeResult> InitializeImplAsync(I
 
             try
             {
-                if (!await RunPackageAnalyzersAsync(context, token).ConfigureAwait(false))
+                if (!await _packageAnalyzer.AnalyzeAsync(context, context.CurrentProject, _analysisState, token).ConfigureAwait(false))
```

Comment:
you're never setting this `_analysisState` and it's overwritten in the caller. Do you want to return the state from the analyze method?

Reply:
Made the method return analysisstate